graphdatascience/system/config_endpoints.py

```diff
@@ -14,15 +14,14 @@ def set(self, key: str, value: Any, username: Optional[str] = None) -> None:
         self._namespace += ".set"
 
         params = {"key": key, "value": value}
+        body = "$key, $value"
 
         # Checking for explicit None as '' is a valid user
         if username is not None:
-            query = f"CALL {self._namespace}($key, $value, $username)"
+            body += ", $username"
             params["username"] = username
```

Comment:
Somehow I think this feels a little bit hacky, with the Cypher parameter placeholder given as a string like this. I think it would be better to have some kind of builder-like class `CypherParamPlaceholders` that is given to the query runner. In this method:
```python
cpp = CypherParamPlaceholders()
cpp.add("key")
...
self._query_runner(...,cpp=cpp,...)
```
Then the in the query runner:
`f"CALL {endpoint}({cpp.build()}) ..."`
or something like this

Reply:
(just an example, might not be the best way)

Reply:
I went with `CallParameters` which is an ordered dict as a builder